### DISCUSSION TITLE
3D Framed Follow Mode Not Following Target

### DISCUSSION THREAD

**Discussion by JoelCreatesGames** [ramokz/phantom-camera#221] [Question / Help Wanted]
It would be desirable to have the 3D framed mode, follow as if it was on a spring arm like the third person mode. For context I'm trying an on-rails shooter where the character itself is moving through the level.

The issue with framed mode is as the ship gets further out it seems to always be in the deadzone and the Distance/Target Offset don't seem to be respected.

I'm not sure if this is a bug, or not implemented yet, either way I am happy to help figure this out and potentially submit a PR

[2024_02_24__framed.webm](https://github.com/ramokz/phantom-camera/assets/62071362/5ab71f9b-3680-4643-890f-57e3662a2f60)

Where in third person mode it follows the ship

[2024_02_24__third.webm](https://github.com/ramokz/phantom-camera/assets/62071362/b09df106-5640-4533-a8a4-2636ca3c3df5)

**Comment by ramokz** [comment]
That looks like a bug.

Is the yellow dot (follow target) being positioned in the expected position in the first example you shared? It seems to be moving exponentially compared to the model.  Haven't done much testing with forward movement for `Framed Follow`, so it's very possible that something isn't being triggered correctly as you touch the `dead zone` limit. Something that should be tested further and fixed as part of #161.

Are you doing any additional logic with the camera beyond what the addon does? Nothing wrong with that, just want to see what variables might be at play.

**Reply by JoelCreatesGames** [reply]
> Is the yellow dot (follow target) being positioned in the expected position in the first example you shared?

yeah it seems to be moving as expected with the controller/ship input

> It seems to be moving exponentially compared to the model

it does seem to be moving quite a bit faster than the ship does

> Something that should be tested further and fixed as part of https://github.com/ramokz/phantom-camera/issues/161.

Let me know if I can help further!

> Are you doing any additional logic with the camera beyond what the addon does?

Nope, using the default phantom camera setup

**Reply by JoelCreatesGames** [reply]
incase this helps, this is my ship's movement

```gdscript
func _physics_process(delta):
	inputVector = Input.get_vector("left", "right", "up", "down")
	
	# Change the velocity and rotation
	velocity.x = move_toward(velocity.x, inputVector.x * MAXSPEED, ACCELERATION)
	velocity.y = move_toward(velocity.y, inputVector.y * MAXSPEED, ACCELERATION)
	velocity.z = -5

	rotation_degrees.z = velocity.x * -2
	rotation_degrees.x = velocity.y / 2
	rotation_degrees.y = -velocity.x / 2
	
	# Apply the movement
	move_and_slide()
	
	# Clamp to within screen bounds
	transform.origin.x = clamp(transform.origin.x, -10, 10)
	transform.origin.y = clamp(transform.origin.y, -6, 6)
```

**Reply by JoelCreatesGames** [reply]
![image](https://github.com/ramokz/phantom-camera/assets/62071362/81df2e02-1c82-425a-b632-0097b5c1c695)


![image](https://github.com/ramokz/phantom-camera/assets/62071362/3dc93899-3641-4b5c-b9d8-35db823634f9)

**Reply by ramokz** [reply]
Think I know what the issue is.

Essentially, the distance is only being calculated and set correctly from the editor, whereas it should really be constantly calculated during runtime. It's something I'm hoping to look into for the refactor, but hadn't come across this particular use case. So the issue you showed here will be great to test against when the work starts on that.
